### ISSUE TITLE
Failed with exit code 139

### ISSUE THREAD

**Issue by R1nc3wind** [marbl/canu#885]
Hi,

I'm trying to assemble some amplicon sequencing data, but I keep getting the exit code 139. The amplicon is only ~700bp long, and I'm running this in an Ubuntu 16.04 virtual environment. The command I've been running (and resulting output) is:

```
./canu -p CLL145_bc04_IgHV_albacore_porechop -d /media/E6DA1377DA13436B/CLL_MinION/CLL_IgH_MinION_Run1/CLL145/ genomeSize=1000 -nanopore-raw /media/E6DA1377DA13436B/CLL_MinION/CLL_IgH_MinION_Run1/CLL145/CLL145_bc04_IgHV_albacore_porechop.fastq merylThreads=4 merylMemory=10 cormhapMemory=10 cormhapThreads=4 obtovlMemory=10 obtovlThreads=4 utgovlMemory=10 utgovlThreads=4 ovsMemory=10 ovsThreads=4 batMemory=10 batThreads=4 gfaMemory=10 gfaThreads=4 minReadLength=100 minOverlapLength=99 stopOnReadQuality=false useGrid=false 
-- Canu snapshot v1.7 +114 changes (r8806 c875536f014fb9096fb2dc14cbf30a2ca09e7bd6)
--
-- CITATIONS
--
-- Koren S, Walenz BP, Berlin K, Miller JR, Phillippy AM.
-- Canu: scalable and accurate long-read assembly via adaptive k-mer weighting and repeat separation.
-- Genome Res. 2017 May;27(5):722-736.
-- http://doi.org/10.1101/gr.215087.116
-- 
-- Read and contig alignments during correction, consensus and GFA building use:
--   Šošic M, Šikic M.
--   Edlib: a C/C ++ library for fast, exact sequence alignment using edit distance.
--   Bioinformatics. 2017 May 1;33(9):1394-1395.
--   http://doi.org/10.1093/bioinformatics/btw753
-- 
-- Overlaps are generated using:
--   Berlin K, et al.
--   Assembling large genomes with single-molecule sequencing and locality-sensitive hashing.
--   Nat Biotechnol. 2015 Jun;33(6):623-30.
--   http://doi.org/10.1038/nbt.3238
-- 
--   Myers EW, et al.
--   A Whole-Genome Assembly of Drosophila.
--   Science. 2000 Mar 24;287(5461):2196-204.
--   http://doi.org/10.1126/science.287.5461.2196
-- 
-- Corrected read consensus sequences are generated using an algorithm derived from FALCON-sense:
--   Chin CS, et al.
--   Phased diploid genome assembly with single-molecule real-time sequencing.
--   Nat Methods. 2016 Dec;13(12):1050-1054.
--   http://doi.org/10.1038/nmeth.4035
-- 
-- Contig consensus sequences are generated using an algorithm derived from pbdagcon:
--   Chin CS, et al.
--   Nonhybrid, finished microbial genome assemblies from long-read SMRT sequencing data.
--   Nat Methods. 2013 Jun;10(6):563-9
--   http://doi.org/10.1038/nmeth.2474
-- 
-- CONFIGURE CANU
--
-- Detected Java(TM) Runtime Environment '1.8.0_162' (from 'java').
-- Detected gnuplot version '5.0 patchlevel 3' (from 'gnuplot') and image format 'png'.
-- Detected 4 CPUs and 10 gigabytes of memory.
-- No grid engine detected, grid disabled.
--
--                            (tag)Concurrency
--                     (tag)Threads          |
--            (tag)Memory         |          |
--        (tag)         |         |          |     total usage     algorithm
--        -------  ------  --------   --------  -----------------  -----------------------------
-- Local: meryl     10 GB    4 CPUs x   1 job     10 GB    4 CPUs  (k-mer counting)
-- Local: cormhap   10 GB    4 CPUs x   1 job     10 GB    4 CPUs  (overlap detection with mhap)
-- Local: obtovl    10 GB    4 CPUs x   1 job     10 GB    4 CPUs  (overlap detection)
-- Local: utgovl    10 GB    4 CPUs x   1 job     10 GB    4 CPUs  (overlap detection)
-- Local: ovb        4 GB    1 CPU  x   2 jobs     8 GB    2 CPUs  (overlap store bucketizer)
-- Local: ovs       10 GB    4 CPUs x   1 job     10 GB    4 CPUs  (overlap store sorting)
-- Local: red        8 GB    4 CPUs x   1 job      8 GB    4 CPUs  (read error detection)
-- Local: oea        4 GB    1 CPU  x   2 jobs     8 GB    2 CPUs  (overlap error adjustment)
-- Local: bat       10 GB    4 CPUs x   1 job     10 GB    4 CPUs  (contig construction with bogart)
-- Local: gfa       10 GB    4 CPUs x   1 job     10 GB    4 CPUs  (GFA alignment and processing)
--
-- Found Nanopore uncorrected reads in the input files.
--
-- Generating assembly 'CLL145_bc04_IgHV_albacore_porechop' in '/media/E6DA1377DA13436B/CLL_MinION/CLL_IgH_MinION_Run1/CLL145'
--
-- Parameters:
--
--  genomeSize        1000
--
--  Overlap Generation Limits:
--    corOvlErrorRate 0.3200 ( 32.00%)
--    obtOvlErrorRate 0.1440 ( 14.40%)
--    utgOvlErrorRate 0.1440 ( 14.40%)
--
--  Overlap Processing Limits:
--    corErrorRate    0.5000 ( 50.00%)
--    obtErrorRate    0.1440 ( 14.40%)
--    utgErrorRate    0.1440 ( 14.40%)
--    cnsErrorRate    0.1920 ( 19.20%)
--
--
-- BEGIN CORRECTION
--
----------------------------------------
-- Starting command on Fri Apr 27 13:53:47 2018 with 1083.131 GB free disk space

    cd .
    /home/Software/canu/Linux-amd64/bin/sqStoreCreate \
      -o ./CLL145_bc04_IgHV_albacore_porechop.seqStore.BUILDING \
      -minlength 100 \
      ./CLL145_bc04_IgHV_albacore_porechop.seqStore.ssi \
    > ./CLL145_bc04_IgHV_albacore_porechop.seqStore.BUILDING.err 2>&1

-- Finished on Fri Apr 27 13:53:50 2018 (3 seconds) with 1083.047 GB free disk space
----------------------------------------

ERROR:
ERROR:  Failed with exit code 139.  (rc=35584)
ERROR:

ABORT:
ABORT: Canu snapshot v1.7 +114 changes (r8806 c875536f014fb9096fb2dc14cbf30a2ca09e7bd6)
ABORT: Don't panic, but a mostly harmless error occurred and Canu stopped.
ABORT: Try restarting.  If that doesn't work, ask for help.
ABORT:
ABORT:   sqStoreCreate failed.
ABORT:
ABORT: Disk space available:  1083.047 GB
ABORT:
ABORT: Last 50 lines of the relevant log file (./CLL145_bc04_IgHV_albacore_porechop.seqStore.BUILDING.err):
ABORT:
ABORT:   
ABORT:   Starting file './CLL145_bc04_IgHV_albacore_porechop.seqStore.ssi'.
ABORT:   
ABORT:     Loading reads from '/media/E6DA1377DA13436B/CLL_MinION/CLL_IgH_MinION_Run1/CLL145/CLL145_bc04_IgHV_albacore_porechop.fastq'
ABORT:       Processed 481855 lines.
ABORT:       Loaded 49411782 bp from:
ABORT:         96371 FASTQ format reads (49411782 bp).
ABORT:       WARNING: 15 reads (0.0156%) with 1281 bp (0.0026%) were too short (< 100bp) and were ignored.
ABORT:   
ABORT:   Finished with:
ABORT:     0 warnings (bad base or qv, too short, too long)
ABORT:   
ABORT:   Loaded into store:
ABORT:     49411782 bp.
ABORT:     96356 reads.
ABORT:   
ABORT:   Skipped (too short):
ABORT:     1281 bp (0.0026%).
ABORT:     15 reads (0.0156%).
ABORT:   
ABORT:   
ABORT:   
ABORT:   Failed with 'Segmentation fault'; backtrace (libbacktrace):
ABORT:   AS_UTL/AS_UTL_stackTrace.C::97 in _Z17AS_UTL_catchCrashiP9siginfo_tPv()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   /usr/include/x86_64-linux-gnu/bits/stdio2.h::98 in fprintf()
ABORT:   stores/sqStoreCreate.C::674 in _Z11createStorePKcjPPcjj()
ABORT:   stores/sqStoreCreate.C::883 in main()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   Segmentation fault (core dumped)
ABORT:
```
Any help is appreciated. Many thanks.

**Comment by skoren** [comment]
Any updates on this?

**Comment by R1nc3wind** [comment]
Thanks for the advice. I sub-sampled 2k reads from the fastq file and changed the overlap and minReadLength to 500. I also downgraded to the 1.7 release and that solved the problem.

**Comment by skoren** [comment]
Thanks for the update, closing since it ran successfully.

**Comment by fbemm** [comment]
I run into something similar right before building the seqStore after oea:

```
    Processed 241749798 lines.
    Loaded 18904388455 bp from:
      2328053 FASTA format reads (18904388455 bp).
    WARNING: 329533 reads (14.1549%) with 157459545 bp (0.8260%) were too short (< 1000bp) and were ignored.

Finished with:
  0 warnings (bad base or qv, too short, too long)

Loaded into store:
  18904388455 bp.
  1998520 reads.

Skipped (too short):
  157459545 bp (0.8260%).
  329533 reads (14.1549%).

Failed with 'Segmentation fault'; backtrace (libbacktrace):
AS_UTL/AS_UTL_stackTrace.C::97 in _Z17AS_UTL_catchCrashiP9siginfo_tPv()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
/usr/include/x86_64-linux-gnu/bits/stdio2.h::98 in fprintf()
stores/sqStoreCreate.C::674 in _Z11createStorePKcjPPcjj()
stores/sqStoreCreate.C::883 in main()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
Segmentation fault
```

Segfault occurs both with 1.7 and snapshot v1.7 +243 changes (r8935 6f3c37525c6bf532be8f585daeaf565507c4c3b1).

**Comment by brianwalenz** [comment]
Thanks for reporting.  Fixed in tip.

**Comment by fbemm** [comment]
Thanks! seqStore is created now but the assembly somehow just stops:

`Unable to assemble uncorrected reads.`

Guess I have to restart the whole thing or at delete parts of the unitiggin.

**Comment by skoren** [comment]
What canu command are you running, it sounds like you're telling it to assemble and not correct the raw reads.

**Comment by fbemm** [comment]
Hell no ;) 

`canu-dev/bin/canu -pacbio-raw $ASM genomeSize=543m -p $OUT -d $DIR gridEngine=sge useGrid=false mhapMemory=64g cnsMemory=64g redMemory=64g batMemory=128g`

**Comment by skoren** [comment]
Ah but if you removed and re-built the seqStore you lost the history of those reads being loaded, this could also be caused by the version change during assembly. Switch to using the trimmed reads as input, something like: ```canu -assemble -pacbio-corrected $DIR/canu-2.00.trimmedReads.fasta.gz```

**Comment by fbemm** [comment]
Makes total sense!
